lopezMidterm/src/reducers/cartReducer.ts

```diff
@@ -0,0 +1,46 @@
+import { ADD_TO_CART, REMOVE_FROM_CART, CLEAR_CART, CartAction, CartItem } from '../types/cartTypes';
+
+interface CartState {
+  items: CartItem[];
+}
+
+const initialState: CartState = {
+  items: [],
+};
+
+const cartReducer = (state = initialState, action: CartAction): CartState => {
```

Comment:
You have a little bit of a mix of Redux-toolkit (RTK) and the older version of Redux. Some of the features will work together so this is okay. In RTK, we don't have to worry about keeping our state immutable because it's done behind the scenes for us.